test/integration/controllermanager/project/project/project_test.go

```diff
@@ -536,5 +537,250 @@ var _ = Describe("Project controller tests", func() {
+				project.Spec.Members = append(project.Spec.Members, gardencorev1beta1.ProjectMember{
+					Subject: rbacv1.Subject{
+						APIGroup: rbacv1.GroupName,
+						Kind:     rbacv1.UserKind,
+						Name:     testUserName,
+					},
+					Role: "viewer",
+				})
+				Expect(testClient.Patch(ctx, project, patch)).To(Succeed())
+
+				By("Ensure new admin has access to NamespacedCloudProfile")
+				Eventually(func(g Gomega) {
+					g.Expect(testUserClient.Get(ctx, namespacedCloudProfileKey, &gardencorev1beta1.NamespacedCloudProfile{})).To(Succeed())
+				}).Should(Succeed())
+
+				By("Ensure viewer cannot update NamespacedCloudProfile")
+				namespacedCloudProfile.Spec.MachineTypes = []gardencorev1beta1.MachineType{
+					{Name: "test-machine-type", CPU: resource.MustParse("2"), Memory: resource.MustParse("1G")},
+				}
+				Expect(testUserClient.Update(ctx, namespacedCloudProfile)).To(BeForbiddenError())
+			})
+
+			It("should allow project admins to read and modify a project's NamespacedCloudProfile for non-special fields", func() {
+				By("Grant project-member respective permissions for NamespacedCloudProfiles")
+				clusterRoleGardenerMember := &rbacv1.ClusterRole{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   "gardener.cloud:system:project-member",
+						Labels: map[string]string{"gardener.cloud/role": "project-member"},
+					},
+					Rules: []rbacv1.PolicyRule{
+						{
+							APIGroups: []string{"core.gardener.cloud"},
+							Resources: []string{"namespacedcloudprofiles"},
+							Verbs:     []string{"get", "list", "watch", "create", "patch", "update", "delete"},
+						},
+					},
+				}
+				Expect(testClient.Create(ctx, clusterRoleGardenerMember)).To(Succeed())
+				DeferCleanup(func() {
+					Expect(testClient.Delete(ctx, clusterRoleGardenerMember)).To(Succeed())
+				})
+
+				By("Add admin to project")
+				patch := client.MergeFrom(project.DeepCopy())
+				project.Spec.Members = append(project.Spec.Members, gardencorev1beta1.ProjectMember{
+					Subject: rbacv1.Subject{
+						APIGroup: rbacv1.GroupName,
+						Kind:     rbacv1.UserKind,
+						Name:     testUserName,
+					},
+					Role: "admin",
+				})
+				Expect(testClient.Patch(ctx, project, patch)).To(Succeed())
+
+				By("Ensure new admin has access to NamespacedCloudProfile")
+				Eventually(func(g Gomega) {
+					g.Expect(testUserClient.Get(ctx, namespacedCloudProfileKey, &gardencorev1beta1.NamespacedCloudProfile{})).To(Succeed())
+				}).Should(Succeed())
+
+				By("Ensure admin without proper role can update NamespacedCloudProfile.Spec.{MachineTypes,VolumeTypes}")
+				namespacedCloudProfile.Spec.MachineTypes = []gardencorev1beta1.MachineType{
+					{Name: "test-machine-type", CPU: resource.MustParse("2"), Memory: resource.MustParse("1G")},
+				}
+				namespacedCloudProfile.Spec.VolumeTypes = []gardencorev1beta1.VolumeType{
+					{Name: "test-volume-type", Class: "standard"},
+				}
+				Expect(testUserClient.Update(ctx, namespacedCloudProfile)).To(Succeed())
```

Comment:
Unlike for other `update` calls, `Eventually` is not used here. Is that intended?

Reply:
If I ensured test user access to the NamespacedCloudProfile earlier already, I omit the `Eventually`, as I really expect it to still work (from authorization point of view).